src/service/layer/trace_err.rs

```diff
@@ -69,7 +68,9 @@ where
         req: Request,
     ) -> impl Future<Output = Result<Self::Response, Self::Error>> + Send + '_ {
         let level = self.level;
-        self.inner.serve(ctx, req).map(move |res| {
+        let serve = self.inner.serve(ctx, req);
```

Comment:
valid but you can just implement it as `async fn serve(...) -> Result<...> { ... }`, is more rama-idiomatic, and easier. Then you do not need the `async move`.